lib/dns/zone.js

```diff
@@ -0,0 +1,458 @@
+    query.sortOrder = query.sort === 'desc' ? 'descending' : 'ascending';
+    delete query.sort;
+  }
+
+  var path = '/managedZones/' + this.name + '/changes';
+
+  this.makeReq_('GET', path, query, true, function(err, resp) {
+    if (err) {
+      callback(err, null, null, resp);
+      return;
+    }
+
+    var changes = (resp.changes || []).map(function(change) {
+      var changeInstance = self.change(change.id);
+      changeInstance.metadata = change;
+      return changeInstance;
+    });
+
+    var nextQuery = null;
+    if (resp.nextPageToken) {
+      nextQuery = extend({}, query, {
+        pageToken: resp.nextPageToken
+      });
+    }
+
+    callback(null, changes, nextQuery, resp);
+  });
+};
+
+/**
+ * Get the metadata for the zone.
+ *
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * zone.getMetadata(function(err, metadata, apiResponse) {
+ *   // Now zone.metadata = metadata
+ * });
+ */
+Zone.prototype.getMetadata = function(callback) {
+  var self = this;
+  var path = '/managedZones/' + this.name;
+
+  this.makeReq_('GET', path, null, true, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    self.metadata = resp;
+
+    callback(null, self.metadata, resp);
+  });
+};
+
+/**
+ * Get the list of records for this zone.
+ *
+ * @param {object=} query - The query object.
+ * @param {string} query.name - Restricts the list to return only records with
+ *     this fully qualified domain name.
+ * @param {string} query.type - Restricts the list to return only records of
+ *     this type. If present, the "name" parameter must also be present.
+ * @param {number} query.maxResults - Maximum number of results to be returned.
+ * @param {string} query.pageToken - The page token.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ *
+ */
+Zone.prototype.getRecords = function(query, callback) {
+  var self = this;
+
+  if (util.is(query, 'function')) {
+    callback = query;
+    query = {};
+  }
+
+  if (util.is(query, 'string')) {
+    this.getRecordsByType_(query, callback);
+    return;
+  }
+
+  var path = '/managedZones/' + this.name + '/rrsets';
+  this.makeReq_('GET', path, query, true, function(err, resp) {
+    if (err) {
+      callback(err, null, null, resp);
+      return;
+    }
+
+    var records = (resp.rrsets || []).map(function(record) {
+      return self.record(record.type, record);
+    });
+
+    var nextQuery = null;
+    if (resp.nextPageToken) {
+      nextQuery = extend({}, query, {
+        pageToken: resp.nextPageToken
+      });
+    }
+
+    callback(null, records, nextQuery, resp);
+  });
+};
+
+/**
+ * @TODO
+ */
+Zone.prototype.import = function(filename, options, callback) {
+  callback(filename, options);
+};
+
+Zone.prototype.record = function(recordType, metadata) {
+  return new Record(this, recordType, metadata);
+};
+
+Zone.prototype.replaceRecords = function(recordType, newRecords, callback) {
```

Comment:
Docstring on this one is going to be important... 

I'm all for this existing (ie, "replace whatever I have of type A with this one"), but it's not atomic -- in that if someone adds a new A record between our getRecords() call and the createChange() call... we'll have some wonky results...